jsonrpc/eth_endpoint.go

```diff
@@ -132,6 +151,19 @@ func (e *Eth) GetBlockByHash(hash types.Hash, fullTx bool) (interface{}, error)
 		return nil, nil
 	}
 
+	// we need to copy it because the store returns header from storage directly
+	// and not a copy, so changing it, actually changes it in storage as well
+	headerCopy := block.Header.Copy()
```

Comment:
please make separate function for filtering block

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1383/commits/bc27867a749f1f8a344958126aaa881399fd0971